kornia/feature/loftr/loftr.py

```diff
@@ -0,0 +1,138 @@
+from typing import Dict, Optional
+
+import torch
+import torch.nn as nn
+
+from .backbone import build_backbone
+from .loftr_module import FinePreprocess, LocalFeatureTransformer
+from .utils.coarse_matching import CoarseMatching
+from .utils.fine_matching import FineMatching
+from .utils.position_encoding import PositionEncodingSine
+
+#from einops.einops import rearrange
+
+
+
+urls: Dict[str, str] = {}
+urls["outdoor"] = "http://cmp.felk.cvut.cz/~mishkdmy/models/loftr_outdoor.ckpt"
+urls["indoor"] = "http://cmp.felk.cvut.cz/~mishkdmy/models/loftr_indoor.ckpt"
+
+default_cfg = {'backbone_type': 'ResNetFPN',
+ 'resolution': (8, 2),
+ 'fine_window_size': 5,
+ 'fine_concat_coarse_feat': True,
+ 'resnetfpn': {'initial_dim': 128, 'block_dims': [128, 196, 256]},
+ 'coarse': {'d_model': 256,
+  'd_ffn': 256,
+  'nhead': 8,
+  'layer_names': ['self',
+   'cross',
+   'self',
+   'cross',
+   'self',
+   'cross',
+   'self',
+   'cross'],
+  'attention': 'linear',
+  'temp_bug_fix': False},
+ 'match_coarse': {'thr': 0.2,
+  'border_rm': 2,
+  'match_type': 'dual_softmax',
+  'dsmax_temperature': 0.1,
+  'skh_iters': 3,
+  'skh_init_bin_score': 1.0,
+  'skh_prefilter': True,
+  'train_coarse_percent': 0.4,
+  'train_pad_num_gt_min': 200},
+ 'fine': {'d_model': 128,
+  'd_ffn': 128,
+  'nhead': 8,
+  'layer_names': ['self', 'cross'],
+  'attention': 'linear'}}
+
+
+class LoFTR(nn.Module):
```

Comment:
add docstring with reference and small snippet usage

Reply:
would be really helpful also to have description of the arguments in the configuration or make a selection with the more relevant ones.

Reply:
Do not review yet :) 
THe PR is far from ready, that is just a cropped version of original repo.

Reply:
ok - mark as draft then :)

Reply:
I thought, that [WIP] is actually the correct mark :)